src/dashboard/pages/login/login_page.py

```diff
@@ -0,0 +1,168 @@
+"""The layout component of the login page."""
+
+import dash
+from dash import callback, dcc, html
+from dash.dependencies import Component, Input, Output, State
+
+PORT = 8000
+ADDRESS = "127.0.0.1"
+PATH = "/login"
+
+dash.register_page(__name__, path=PATH, nav_item=False)
+
+# sets the colors of the login page
+colors = {
+    "background": "#E9E9F2",
+    "text": "#7FDBFF",
+    "meny_back": "#636AF2",
+    "white": "#FFFFFF",
+    "dark_purp": "#2F3273",
+    "black": "#00000",
+}
+
+NORMAL_FIELD_CLASS = "w-[420px] p-[13px] rounded-full"
+
+ERROR_FIELD_CLASS = (
+    "w-[420px] p-[13px] bg-red-50 "
+    "placeholder-red-700 rounded-full focus:ring-red-500"
+    "focus:border-red-500 dark:text-red-500 "
+    "dark:placeholder-red-500 dark:border-red-500"
+)
+
+PASS_FIELD_EXTRA_CLASS = " my-4"
+
+
+def layout() -> Component:
```

Comment:
Decrease nesting in this function. 3-4 levels of indentation is usually a good target. Maybe extract logical parts of the layout to other functions, or separate assignments.